### ISSUE TITLE
Strange echo output (value to be echoed is not the same with it's real value)

### ISSUE THREAD

**Issue by gordianknot1981** [nim-lang/Nim#1338]
```
output
========================================
------------------------------------------------------------------
var parent = "root"
 echo parent: root, evaluate parent == "root":  true
 echo "root": root
------------------------------------------------------------------
var parent = "root"
 echo parent: messy, evaluate parent == "root":  true  # instead of it's real value "root", messy was displayed
 echo "root": messy                                    # echo a pure string "root", got messy too
=========================================

```
above output is generated from my following code, it summaries the strange condition I encountered.

Assuming 'parent' is "root"
if I echo 'parent' I got "Tittle" instead of it's real value "root". To evaluate parent == "root" I got true. It's quite beyond my comprehension of why the value to be echoed is not the same with it's real value. By the way, any slight modification to my code seems clear the problem ( even if I just drop a  irrelevant echo expression, a empty proc,  or a unused variable ).

see the code ..
(Pardon for my messy code. I try to simplify the code to produce a bug trigger, but have run out of my mind of how to reproduce it in a simplified way. )
```
iimport macros


macro Kivy*(body: stmt): stmt {.immediate.} =
   proc head  ( x:PNimrodNode):PNimrodNode {.inline.}= x[0]
   proc tail  ( rest:PNimrodNode):PNimrodNode = 
      var result:PNimrodNode = newNimNode(nnkRecList)
      if rest.len > 1: 
         for i in 1..rest.len-1:
            result.add(rest[i])
      return result
 
   proc emptyProc(parent,cls, base:string, ret: var PNimrodNode): void = 
      discard

   proc parseNode [T](node:T, lid:T, lnod:T,ret: var T, parent:string, flag:string, counter = 0) = 
      var x  = head node
      var xs = tail node

      # replace head node with node[0]; problem cleared
      #-----------------------------------------------
      # var x      = node[0]       Replace "head node" with "node[0]" seems clear the problem
      # var xs     = tail node     But what if we echo repr(x) and repr(xs) at following line?
      # echo repr(x)
      # echo repr(xs)              # <= problem showup again


      # replace "tail node" with inline code( or rewirte proc with template); problem cleared
      # Seems totally clear the problem, but not sure where the key is?
      #-----------------------------------------------
      # var rest:PNimrodNode = newNimNode(nnkRecList)
      # if node.len > 1: 
      #    for i in 1..node.len-1:
      #       rest.add(node[i])
      # var xs = rest


      case x.kind:
         of nnkBracketExpr:
            var base   = x[1]      
            var cls    = x[0]       
            var unusded= "str"     # drop this line, problem cleared
            var parent = "messy"   
            var x      = newNimNode(nnkCall).add(ident(repr(cls)))

            
            emptyProc("root", repr(cls), repr(base), ret)      

            # replace emptyProc's first arg "root" with any other strings, problem cleared
            #----------------------------------------------------------------------------
            # emptyProc("anyStringOtherThanRoot", repr(cls), repr(base), ret)      


            # xs.len is always bigger than zero, so it seems if statement is irrelevant
            # to the problem, but what if we drop this if statement?
            if xs.len > 0: 
               parseNode(x,lid,lnod,ret, parent,flag,counter)
               parseNode(xs,lid,lnod,ret, parent,flag,counter)
               return

            # drop if statement, problem cleared:
            #-----------------------------------
            #parseNode(x,lid,lnod,ret, parent,flag,counter)
            #parseNode(xs,lid,lnod,ret, parent,flag,counter)
            #return 

         else:
            return
   
   echo "========================================"
   var ret = newNimNode(nnkRecList)
   for node in body.children:
      var parent = "root"
      echo "------------------------------------------------------------------"
      echo "var parent = \"root\""
      echo " echo parent: ",parent,", evaluate parent == \"root\":  ", parent == "root"
      echo " echo \"root\": ", "root"
      parseNode(node, nil, nil, ret, parent, "str", 0)
      
   echo "========================================="


Kivy:
   Title[Label]:                 
      font: "title_font"         
                            
   Sub[DivLayout]:               
      size_hint: [None,None]     

```

**Comment by jangko** [comment]
this already fixed, no longer valid issue for nim 0.12.0

**Comment by reactormonk** [comment]
Can confirm.
